src/battle_script_commands.c

```diff
@@ -16524,6 +16525,22 @@ void BS_TryDefog(void)
     }
 }
 
+void BS_TryUpperHand(void)
+{
+	NATIVE_ARGS(const u8 *failInstr);
+	
+	if (gProtectStructs[gBattlerTarget].obstructed) // Fails if target is obstructed
+		gBattlescriptCurrInstr = cmd->failInstr;
+	else if (GetBattlerTurnOrderNum(gBattlerAttacker) > GetBattlerTurnOrderNum(gBattlerTarget)) // Fails if user moves after target
+		gBattlescriptCurrInstr = cmd->failInstr;
+	else if (IS_MOVE_STATUS(gBattleMons[gBattlerTarget].moves[gBattleStruct->chosenMovePositions[gBattlerTarget]])) // Fails if target is using a status move
+		gBattlescriptCurrInstr = cmd->failInstr;
+	else if (GetChosenMovePriority(gBattlerTarget) <= 0) // Fails if priority is not greater than 0
+		gBattlescriptCurrInstr = cmd->failInstr;
+	else
```

Comment:
Missing check for failing if target's priority move is greater than 3

Reply:
Is that not covered in "if the target has already moved"?

Reply:
No, because there are other scenarios where the move order can get shuffled around (eg. After You)